### DISCUSSION TITLE
How to find Gal.putImageBytes() saved file path & name once saved to photo gallery

### DISCUSSION THREAD

**Discussion by wwyung** [natsuk4ze/gal#224] [Q&A]
### Did you check the documentation and it did not help?

- [X] I have carefully read the [README](https://github.com/natsuk4ze/gal/blob/main/README.md)
- [X] I have carefully searched the [existing documentation](https://github.com/natsuk4ze/gal/wiki)
- [X] I have carefully searched the [existing answer](https://github.com/natsuk4ze/gal/discussions)
- [X] I have carefully read the [contribution guide](https://github.com/natsuk4ze/gal/blob/main/CONTRIBUTING.md)

### Descrpition

Hi, here is how I save a camera captured pictured via image picker, and I then use Gal to save to photo gallery. My question are:

1 - I didn't specify the album and pic name when calling putImageBytes(), I assume Gal will generate an unique name that will not overwriting any existing photos already existed in the photo gallery in this case, right?

2 - I want to find out new file path & name after puImageBytes() completed successfully, how do I do that?

      ImagePicker _imagePicker = ImagePicker();
      XFile? returnedImage = await _imagePicker.pickImage(source: ImageSource.camera);

      if (returnedImage != null) {
        File imageFile = File(returnedImage.path);
        Uint8List imageBytes = await returnedImage.readAsBytes();
        await Gal.putImageBytes(imageBytes);
    }

**Comment by natsuk4ze** [comment]
There is currently no way to receive the file path or name after saving an image to the gallery. Detailed use cases and good motivation will be on the list of new features to consider.

**Comment by thanglq1** [comment]
I also need to return the album path to display all images in 'My Album' inside my app, because I save a lot of images into 'My Album', and I want users to be able to manage their images inside my app instead of them opening the gallery app.

**Reply by thanglq1** [reply]
I think I will use ```await Gal.putImage(imagePath, album: albumName);``` for my case

**Comment by dryasbk** [comment]
It would be very nice to add name for the image/ video and get the new path of the saved media.

**Comment by natsuk4ze** [comment]
https://github.com/natsuk4ze/gal/issues/232

**Comment by natsuk4ze** [comment]
Here is why I am reluctant to work on returning file path.

- There are other ways to achieve the use case
- In order to get a pass on iOS, I would need full access to all photos, not just write permissions
- On iOS, getting the path to a file and manipulating it directly is deprecated
- The path can be changed by the user or the system without the app's intention -> it only can be used as a temporary file path, so if that is the case, the path of a file already stored locally is sufficient
